pyam/core.py

```diff
@@ -2340,15 +2340,20 @@ def concat(dfs, ignore_meta_conflict=False, **kwargs):
             f"you passed an object of type '{dfs.__class__.__name__}'!"
         )
 
-    # cast first element in list to IamDataFrame (if necessary)
-    df = dfs[0] if isinstance(dfs[0], IamDataFrame) else IamDataFrame(dfs[0], **kwargs)
+    dfs_iter = iter(dfs)
+
+    # cast to IamDataFrame if necessary
+    def as_iamdataframe(df):
+        return df if isinstance(df, IamDataFrame) else IamDataFrame(df, **kwargs)
+
+    df = as_iamdataframe(next(dfs_iter))
```

Comment:
If one wants to catch the StopIteration, the easiest is:

```suggestion
    df = next(dfs_iter, None)
    if df is None:
        raise ValueError("No objects to concatenate")
    df = as_iamdataframe(df)
```

but I guess that becomes more and more difficult to understand without any clear gain on the other side, let me rewrite it back to list.

Reply:
thanks!